### PR TITLE
Support `mut` parameters on `#[ffi_export]`ed functions

### LINKED ISSUE #110: `#[ffi_export]` doesn't work with `mut` keyword in parameters
This code:
```rust
#[ffi_export]
pub fn f(mut v: repr_c::Vec<u8>) {
    // let mut v = v;
    v.with_rust_mut(|v| *v = Vec::new());
}
```
Fails with this error:
image

This error goes away when uncommenting the line and removing `mut` from the parameter

### PR DESCRIPTION
**Pull request by danielhenrymantilla**
- Also bumps MSRV due to a `libc` compilation bug on macos

Fixes #110

### CODE REVIEW

File: ffi_tests/generated.h [hunk 1]
```diff
@@ -28,9 +28,27 @@ int32_t read_foo (
...
+#endif
+;
+
+void check_bar (
+    Bar_t _bar);
+
 /** \brief
  *  Concatenate the two input strings into a new one.
- *
+ *
```
Comment (hamchapman) [inline ffi_tests/generated.h:51]: Whitespace always finds a way back in

Comment (danielhenrymantilla) [reply ffi_tests/generated.h:51]: Yeah this is on the `master` version which does not have your fix 😄

Comment (hamchapman) [reply ffi_tests/generated.h:51]: Ah, makes sense! Ignore me :homer_reverse_into_hedge:

Comment (danielhenrymantilla) [reply ffi_tests/generated.h:51]: 😆 I really need to get all these branches re-unified 😩

**Pull request merged**